spec/book_spec.rb

```diff
@@ -0,0 +1,40 @@
+require_relative '../book'
+require_relative '../rental'
+require_relative '../person'
+
+describe Book do
+  before :each do
+    @book1 = Book.new('lard of the rings', 'martin')
+    @book2 = Book.new('game of throne', 'george')
+    @person = Person.new(98, name: 'Rosy')
+  end
+
+  context 'When creating a new book' do
+    it 'should have the following attributes (title, author, and rentals)' do
+      expect(@book1.title).to eql 'lard of the rings'
+      expect(@book1.author).to eql 'martin'
+      expect(@book1.rentals.count).to eq(0)
+    end
+    it 'should be an instance of the class Book' do
+      expect(@book1).to be_instance_of Book
+      expect(@book1).to be_instance_of Book
+    end
+  end
+
+  context 'when adding a rental for a Book' do
+    it 'can add a rental' do
+      expect(@book1).to respond_to(:add_rental)
+    end
+    it 'has it\'s rentals array length increased' do
+      expect(@book1.rentals.count).to eq(0)
+      Rental.new(Time.now, @person, @book1)
+      expect(@book1.rentals.count).to eq(1)
+    end
+    it 'can add a new rental with add_rental method' do
+      expect(@book2.rentals.count).to eq(0)
+      rental = Rental.new(Time.new, @person, @book1)
+      @book2.add_rental(rental)
+      expect(@book2.rentals.count).to eq(1)
+    end
+  end
+end
```

Comment:
- [ ] Great job so far @AJ193 and @sunga12 There are a few changes I will suggest you implement and you'll be  on your way to the next MIlestone :wink: 
- [ ] After running all your tests, I noticed that 2 of your tests are failing. So kindly fix those as well so that all your tests can pass as expected :100: 

- Failing Tests
![image](https://github.com/AJ193/OOP-school-library/assets/102699958/b2e5f431-4766-4da2-929c-a9102be5a92e)